raw_linux.go

```diff
@@ -279,6 +279,34 @@ func (p *packetConn) SetBPF(filter []bpf.RawInstruction) error {
 	)
 }
 
+// SetPromisc enables/disables interface promiscuous mode through syscall
+// when thread exits interface returns to previous state
+// 0 == disabled, 1 == enabled
+type packet_mreq struct {
+	mr_ifindex int32
+	mr_type    uint16
+	mr_alen    uint16
+	mr_address [8]byte
+}
+
+func (p *packetConn) SetPromisc(m int) error {
+	mreq := packet_mreq{
+		mr_ifindex: int32(p.ifi.Index),
+		mr_type:    syscall.PACKET_MR_PROMISC,
```

Comment:
Let's stick with https://godoc.org/golang.org/x/sys/unix constants. BSD uses `syscall` because some of the BPF stuff isn't ported.